Source/NSCollectionView.m

```diff
@@ -1114,4 +1199,382 @@ - (BOOL) wantsPeriodicDraggingUpdates
   return YES;
 }
 
+/* New methods for later versions of macOS */
+
+// 10.11 methods...
+
+/* Locating Items and Views */
+
+- (NSArray *) visibleItems
+{
+  return _visibleItems;
+}
+
+- (NSSet *) indexPathsForVisibleItems
+{
+  return _indexPathsForVisibleItems;
+}
+
+- (NSArray *) visibleSupplementaryViewsOfKind: (NSCollectionViewSupplementaryElementKind)elementKind
+{
+  return [_visibleSupplementaryViews objectForKey: elementKind];
+}
+
+- (NSSet *) indexPathsForVisibleSupplementaryElementsOfKind: (NSCollectionViewSupplementaryElementKind)elementKind
+{
+  return nil;
+}
+
+- (NSIndexPath *) indexPathForItem: (NSCollectionViewItem *)item
+{
+  return nil;
+}
+
+- (NSIndexPath *) indexPathForItemAtPoint: (NSPoint)point
+{
+  return nil;
+}
+
+- (NSCollectionViewItem *) itemAtIndexPath: (NSIndexPath *)indexPath
+{
+  return nil;
+}
+
+- (NSView *)supplementaryViewForElementKind: (NSCollectionViewSupplementaryElementKind)elementKind 
+                                atIndexPath: (NSIndexPath *)indexPath
+{
+  return nil;
+}
+
+- (void) scrollToItemsAtIndexPaths: (NSSet *)indexPaths 
+                    scrollPosition: (NSCollectionViewScrollPosition)scrollPosition
+{
+}
+
+/* Creating Collection view Items */
+
+- (NSCollectionViewItem *) makeItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier 
+                                     forIndexPath: (NSIndexPath *)indexPath
+{
+  return nil;
+}
+
+- (void) registerClass: (Class)itemClass 
+ forItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier
+{
+  [self registerClass: itemClass
+        forSupplementaryViewOfKind: GSNoSupplementaryElement
+       withIdentifier: identifier];
+}
+
+- (void) registerNib: (NSNib *)nib 
+         forItemWithIdentifier: (NSUserInterfaceItemIdentifier)identifier
+{
+  [self registerNib: nib
+        forSupplementaryViewOfKind: GSNoSupplementaryElement
+     withIdentifier: identifier];
```

Comment:
Strange indentation

Reply:
It seems to be how emacs wanted to place it, but I have aligned it.